### DISCUSSION TITLE
Specify GPU

### DISCUSSION THREAD

**Discussion by Onyoursix** [pyannote/pyannote-audio#1108] [Q&A]
My box has multiple GPUs, is there a way to specify which GPU I want to process on?

**Comment by hbredin** [comment]
Setting `CUDA_VISIBLE_DEVICES` environment variable should do the trick.
See [this](https://developer.nvidia.com/blog/cuda-pro-tip-control-gpu-visibility-cuda_visible_devices/) and [this](https://pytorch.org/docs/stable/notes/cuda.html#device-agnostic-code).

**Comment by hbredin** [comment]
[This PR](https://github.com/pyannote/pyannote-audio/pull/1204) adds a new `pipeline.to(device)` API. Would you mind testing it?
